packages/react-router-dom/index.tsx

```diff
@@ -403,40 +413,45 @@ export const NavLink = React.forwardRef<HTMLAnchorElement, NavLinkProps>(
     },
     ref
   ) {
-    let location = useLocation();
     let path = useResolvedPath(to);
+    let match = useMatch({ path: path.pathname, end, caseSensitive });
```

Comment:
@mjackson Do you foresee any issue switching the internal `isActive` implementation to use the implementation suggested in the [example](https://github.com/remix-run/react-router/blob/main/examples/custom-link/src/App.tsx#L37) for folks who want a custom implementation?  All tests still pass - but I wasn't sure if there was a historical reason for the manual calculation internally.